Plugins/Detray/include/Detray.hpp

```diff
@@ -0,0 +1,552 @@
+///
+/// @note return -1 if not found, to be interpreted by the caller
+int findVolume(
+    const Acts::Experimental::DetectorVolume* volume,
+    const std::vector<const Acts::Experimental::DetectorVolume*>& volumes) {
+  auto candidate = std::find(volumes.begin(), volumes.end(), volume);
+  if (candidate != volumes.end()) {
+    return std::distance(volumes.begin(), candidate);
+  }
+  return -1;
+}
+}  // namespace
+
+namespace detray{
+
+    /// These functions perform the conversion of the ACTS detector to the detray detector. 
+    /// The payloads are created and populated by traversing the initial ACTS detector object.
+
+    /// detray geometry writer function, debug purposes 
+    void detray_detector_print(const detector_t& det){
+
+        std::ofstream outputFile("data_try.json");
+        nlohmann::ordered_json out_json;
+        typename detector_t::name_map names{};
+        out_json["data"] = detray::io::geometry_writer::convert(det, names);
+        outputFile << out_json << std::endl;
+        return;
+    }
+    
+    /// @return the transform_payload(translation, rotation) of each surface/volume
+    /// @param Transform3 acts object, Transform3JsonConverter::Options
+    /// @brief convert the acts transform to detray transform payload
+    static io::transform_payload detray_converter_transf(
+        const Transform3& t, const Transform3JsonConverter::Options& options){
+        //nlohmann::json Acts::Transform3JsonConverter::toJson(const Transform3& t, const Transform3JsonConverter::Options& options)
+
+        io::transform_payload p_acts;
+
+        auto translation = t.translation();
+        if (translation != Acts::Vector3(0., 0., 0) || options.writeIdentity) {
+            std::array<Acts::ActsScalar, 3> tdata = {translation.x(), translation.y(),
+                                                    translation.z()};
+            p_acts.tr = tdata;
+        } else {
+            p_acts.tr = {};
+        }
+        auto rotation =  options.transpose ? t.rotation().transpose() : t.rotation();
+        std::array<Acts::ActsScalar, 9> rdata;
+        if (rotation != Acts::RotationMatrix3::Identity() || options.writeIdentity) {
+            rdata = {
+                rotation(0, 0), rotation(0, 1), rotation(0, 2),
+                rotation(1, 0), rotation(1, 1), rotation(1, 2),
+                rotation(2, 0), rotation(2, 1), rotation(2, 2)};
+            p_acts.rot = rdata;
+        } else {
+            p_acts.rot = {};
+        }
+
+        return p_acts;
+    }
+
+    /// @return the mask_payload of the surface @param bounds, @param portal
+    /// @brief convert the acts bounds to detray mask payload
+    static io::mask_payload detray_converter_mask(
+        const Acts::SurfaceBounds& bounds, bool portal){
+        //Acts::SurfaceBoundsJsonConverter::toJsonDetray
+
+        detray::io::mask_payload mask_pd;
+        auto [shape, boundaries] = DetrayJsonHelper::maskFromBounds(bounds, portal);
```

Comment:
Probably we should shift the  `DetrayJsonHelper::maskFromBound` code to here and call it from the Json helper.